File: 257-binary-tree-paths/257-binary-tree-paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    res = []
    def path(self,root,curr):
        if not root: return
        if not root.left and not root.right:
            curr += str(root.val)
            self.res.append(curr)
            return
        curr += str(root.val) + "->"
        if(root.left): self.path(root.left,curr)
        if(root.right): self.path(root.right,curr)
    
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        self.res = []
        self.path(root,"")
        return self.res

File: 257-binary-tree-paths/257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void path(TreeNode* root, vector<string>& res, string curr){
        if(!root) return;
        if(!root->left && !root->right){
            curr += to_string(root->val);
            res.push_back(curr);
            return;
        }
        curr += to_string(root->val)+"->";
        if(root->left) path(root->left,res,curr);
        if(root->right) path(root->right,res,curr);
    }
    
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> res;
        path(root,res,"");
        return res;
    }
};